src/main/java/com/commercetools/sync/commons/utils/SphereClientConfigOptions.java

```
@@ -0,0 +1,107 @@
+        this.clientConfig = clientConfig;
+        this.timeout = timeout;
+        this.initialDelay = initialDelay;
+        this.timeUnit = timeUnit;
+        this.maxRetries = maxRetries;
+        this.maxParallelRequests = maxParallelRequests;
+    }
+
+    public SphereClientConfig getClientConfig() {
+        return clientConfig;
+    }
+
+    public TimeUnit getTimeUnit() {
+        return timeUnit;
+    }
+
+    public long getTimeout() {
+        return timeout;
+    }
+
+    public long getInitialDelay() {
+        return initialDelay;
+    }
+
+    public int getMaxRetries() {
+        return maxRetries;
+    }
+
+    public int getMaxParallelRequests() {
+        return maxParallelRequests;
+    }
+
+    public static class Builder {
+        private SphereClientConfig clientConfig;
+        protected long timeout;
+        protected long initialDelay;
+        protected TimeUnit timeUnit;
+        protected int maxRetries;
+        protected int maxParallelRequests;
+
+        public Builder(SphereClientConfig clientConfig) {
+            this.clientConfig = clientConfig;
+            this.timeout = DEFAULT_TIMEOUT;
+            this.timeUnit = DEFAULT_TIMEOUT_TIME_UNIT;
+            this.initialDelay = INITIAL_RETRY_DELAY;
+            this.maxRetries = MAX_RETRIES;
+            this.maxParallelRequests = MAX_PARALLEL_REQUESTS;
+        }
+
+        public Builder withTimeout(long timeout) {
+            this.timeout = timeout;
+            return this;
+        }
+
+        public Builder withInitialDelay(long initialDelay) {
+            if(initialDelay < timeout) {
+                this.initialDelay = initialDelay;
+            }
+            return this;
+        }
+
+        public Builder withTimeUnit(TimeUnit timeUnit) {
+            this.timeUnit = timeUnit;
+            return this;
+        }
+
+        public Builder withMaxRetries(int maxRetries) {
+            if(maxRetries > 1) {
```

please throw an IllegalArgumentException instead of silently omitting the value, same for your other validations.

---

Maybe you could create a validation (not negative number etc.) for all the builder methods?

---

added IllegalArguementException. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/fa9d2855b987d5483afec9eb7f49dbcd7d2cd45b